### ISSUE TITLE
[FEATURE REQUEST]: tabs rather than separate windows

### ISSUE THREAD

**Issue by DutchPete** [typora/typora-issues#2757]
If I open more than 1 file with Typora, each file is opened in a separate instance of Typora, i.e. in a separate window.

Moving from 1 file to the next means having to run past other windows before getting to the relevant Typora window. It would be a lot more efficient and pleasant if files are opened in tabs so one can just go directly to the relevant tab. A convenient side feature would be to have keyboard shortcuts to move from 1 tab to the next - e.g. <kbd>CTRL</kbd>+<kbd>PgUp/Dn</kbd>, etc.

**Comment by vassudanagunta** [comment]
See #1000

**Comment by DutchPete** [comment]
@vassudanagunta : thanks for linking to issue #1000. I upvoted it, although you only speak about macOS, whereas I use Linux Mint. If #1000 is implemented it'll probably work on the other platforms too, but with this reply I want to make sure Abner takes Linux into consideration too.

**Comment by abnerlee** [comment]
duplicate with #1000 #638

**Comment by DutchPete** [comment]
> duplicate with #1000 #638

@abnerlee : I am really baffled. This issue is important to a lot of users, yet you keep closing them.
#638 is 2 years old, #1000 is 1 year old, yet the feature only exists in the mac version.
Please, what is your thinking on this ???

**Comment by DutchPete** [comment]
@vassudanagunta : thanks for clarifying that he is consolidating issues.

As for the rest of your comment: I am aware that @abnerlee is a 1-man shop providing a free app, and that I am not obliged to use it. I merely asked him politely his thinking on this issue because it has been outstanding for quite a while and many people are in favour of it.

I did not mean to insult or harass @abnerlee because he has done a fantastic job, AND I have drawn his attention several times to forums where I praised his app very highly. I hope that makes my side a bit clearer to you. So, please, do not lecture me - since he is a 1-man shop I doubt you are his authorised spokesperson. Thank you.

**Comment by vassudanagunta** [comment]
um, i was "lecturing" you because you came off as "lecturing" him. 🤷🏾‍♂️

reread your "I am really baffled" comment, or ask a friend of yours to read it and give you an honest opinion.

**Comment by DutchPete** [comment]
@vassudanagunta : if "I am really baffled" is "lecturing" to you, you need to open a dictionary to learn what it means. Furthermore, saying that this feature is important to a lot of users is not "lecturing", but merely pointing out that it is a feature many users have upvoted. Since this request has been outstanding for quite a while I politely asked him if there is a timeline.

I'll grant you that pointing out @abnerlee is not closing the issue repeatedly but consolidating requests, which I had not realised, was the right thing to do and I am fine with t. You need not have gone any further than that.

So, unless you are @abnerlee's authorised spokesperson or his authorised moderator, it is, to put it politely, not your business to tell me how to write or what to say. @abnerlee is undoubtedly capable of doing that himself. And I certainly don't need you to point out the app is free of charge at this stage so I should shut up and be grateful. @abnerlee is still developing the app, so he is entitled to input from users - if I am not mistaken you also provide him with input, and I do not think you are the only one entitled to do so.

If I have something to say I will do so, whether you like it or not, agree or not. If you don't like my comments don't read them.

**Comment by vassudanagunta** [comment]
I meant the *entirety* of your "I am really baffled" comment. It's not *what* you're pointing out, it's *the tone of your words*. It sure didn't sound "polite". Note that I said "came off".  I'm pretty sure more people would agree with me. I even asked you to ask a friend.

This is a community, a public space. It is entirely my business to point out unfair or unkind words if I so choose. As you say, "If you don't like my comments don't read them."  But a person who claims to be kind wouldn't so categorically reject feedback the way you have. Kindness goes hand-in-hand with humility. Kindness is not defensive, it listens.

Anyway, this is my last word.

**Comment by DutchPete** [comment]
@vassudanagunta: the tone of my words is what you read into it, that is entirely subjective. If it did not sound polite to you, that is your opinion, FWIW. @abnerlee has not reacted negatively to my initial at the top of this page, so the person who is actually concerned, is OK with it.

No, I have no problem with feedback. I have a problem with your type of "feedback", and whether you like it or not, agree or not, I don't take kindly to it because, apart from its tone (subjective, I know), it is entirely misplaced. I admitted to your gentle correction about the consolidation, and I have admitted mea culpa. But the rest of your comment was entirely uncalled for. Goodbye.
